proxy/src/bin/proxy.rs

```diff
@@ -92,6 +93,27 @@ struct ProxyCliArgs {
     /// Require that all incoming requests have a Proxy Protocol V2 packet **and** have an IP address associated.
     #[clap(long, default_value_t = false, value_parser = clap::builder::BoolishValueParser::new(), action = clap::ArgAction::Set)]
     require_client_ip: bool,
+    /// Rate limit algorithm. Makes sense only if `disable_rate_limiter` is `false`.
+    #[clap(value_enum, long, default_value_t = proxy::rate_limiter::RateLimitAlgorithm::Aimd)]
+    rate_limit_algorithm: proxy::rate_limiter::RateLimitAlgorithm,
+    /// Initial limit for dynamic rate limiter. Makes sense only if `rate_limit_algorithm` is *not* `None`.
+    #[clap(long, default_value_t = 10)]
+    initial_limit: usize,
+    /// Minimum limit for AIMD algorithm. Makes sense only if `rate_limit_algorithm` is `Aimd`.
+    #[clap(long, default_value_t = 1)]
+    aimd_min_limit: usize,
+    /// Maximum limit for AIMD algorithm. Makes sense only if `rate_limit_algorithm` is `Aimd`.
+    #[clap(long, default_value_t = 1000)]
+    aimd_max_limit: usize,
```

Comment:
Hmm, my initial reaction is that 1000 is not very much for a maximum. But I'm not sure. How many concurrent requests can one control plane instance handle? What about all the control plane instances altogether?

Reply:
Hm, from [grafana dashboard](https://neonprod.grafana.net/d/S3MPUUonz/neon-console?orgId=1&refresh=5m) I see that ~500 goroutines is about 100% of CPU utilization for the console.
But there is also [grafana dashboard](https://neonprod.grafana.net/d/b74c1a8d-aa7e-48e6-a951-c92a2dde0996/neon-control-plane?orgId=1&refresh=5m) that shows that ~50 goroutines is about 1.5% of CPU utilization for the control plane.

I am a bit confused about that.

Probably @ololobus has a better understanding what is the right data.

Reply:
> But there is also [grafana dashboard](https://neonprod.grafana.net/d/b74c1a8d-aa7e-48e6-a951-c92a2dde0996/neon-control-plane?orgId=1&refresh=5m) that shows that ~50 goroutines is about 1.5% of CPU utilization for the control plane.

> I am a bit confused about that.

These new control-planes are mostly idle, it's better to look on console dashboard for now

Reply:
> Hmm, my initial reaction is that 1000 is not very much for a maximum. But I'm not sure. How many concurrent requests can one control plane instance handle? What about all the control plane instances altogether?

It's hard to answer right away, because we didn't do benchmarks / cap tests in close to prod env, but on my M2 laptop with Postgres in Docker one cplane instance gives
- 10k RPS for ProxyGetRoleSecret
- 200-300 RPS for ProxyWakeCompute

Both done from 200 clients with the same endpoint_id